libqtile/core/manager.py

```diff
@@ -326,39 +329,66 @@ async def finalize(self):
             logger.exception('exception during finalize')
             self.core.remove_listener()
 
-    def _process_fake_screens(self):
+    def _process_fake_screens(self) -> List[Screen]:
         """
         Since Xephyr and Xnest don't really support offset screens, we'll fake
         it here for testing, (or if you want to partition a physical monitor
         into separate screens)
         """
+        screens = []
+
         for i, s in enumerate(self.config.fake_screens):
-            # should have x,y, width and height set
+            # should have x, y, width and height set
             s._configure(self, i, s.x, s.y, s.width, s.height, self.groups[i])
             if not self.current_screen:
                 self.current_screen = s
-            self.screens.append(s)
+            screens.append(s)
 
-    def _process_screens(self) -> None:
+        return screens
+
+    def _process_screens(self) -> List[Screen]:
         if hasattr(self.config, 'fake_screens'):
-            self._process_fake_screens()
-            return
+            return self._process_fake_screens()
 
+        screens = []
         screen_info = self.core.get_screen_info()
 
         for i, (x, y, w, h) in enumerate(screen_info):
-            if i + 1 > len(self.config.screens):
-                scr = Screen()
-            else:
+            if i < len(self.screens):
+                scr = self.screens[i]
+            elif i < len(self.config.screens):
                 scr = self.config.screens[i]
+            else:
+                scr = Screen()
 
             if not self.current_screen:
                 self.current_screen = scr
 
             scr._configure(
                 self, i, x, y, w, h, self.groups[i],
             )
-            self.screens.append(scr)
+            screens.append(scr)
+
+        return screens
```

Comment:
Why not modifying `self.screens` directly?

Reply:
Hmm it made sense while I was hooking it up with _process_screens but rethinking it, some of these changes might not be necessary.